### PR TITLE
Flutter gallery dialog demos

### LINKED ISSUE #1564: Gallery App has Dialogs
From https://www.google.com/design/spec/components/dialogs.html#

- [ ] Alerts w/out title bars
- [ ] Alerts w/ title bars
- [ ] Simple
- [ ] Confirmation
- [ ] Full-screen

### PR DESCRIPTION
**Pull request by HansMuller**
Fixes https://github.com/flutter/flutter/issues/1564

### CODE REVIEW

File: examples/material_gallery/lib/demo/buttons_demo.dart [hunk 1]
```diff
@@ -122,13 +122,7 @@ class _ButtonsDemoState extends State<ButtonsDemo> {
...
               child: new Text("Launch Demo"),
               onPressed: () {
-                showDialog(context: context, child: new DialogDemo()).then((String value) {
-                  if (value != null) {
-                    Scaffold.of(context).showSnackBar(new SnackBar(
-                      content: new Text('You dismissed the dialog with "$value"')
-                    ));
-                  }
-                });
+                Navigator.push(context, new MaterialPageRoute(builder: (_) => new DialogDemo()));
```
Comment (Hixie) [inline examples/material_gallery/lib/demo/buttons_demo.dart]: If possible avoid using `_` in sample code

File: examples/material_gallery/lib/demo/full_screen_dialog_demo.dart [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+      time = new TimeOfDay(hour: dateTime.hour, minute: dateTime.minute),
+      super(key: key) {
+    assert(onChanged != null);
+  }
+
+  final DateTime date;
+  final TimeOfDay time;
+  final ValueChanged<DateTime> onChanged;
+
+  // Add an onChanged() method. Caller is responsible for rebuilding with new values.
```
Comment (Hixie) [inline examples/material_gallery/lib/demo/full_screen_dialog_demo.dart]: TODO()

File: examples/material_gallery/lib/demo/full_screen_dialog_demo.dart [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+  // DOn't need to setState here so this component can be stateless
```
Comment (Hixie) [inline examples/material_gallery/lib/demo/full_screen_dialog_demo.dart]: typo. Though I think that whole line can probably be removed.

**Comment by Hixie** [comment]
lgtm.

File: examples/material_gallery/lib/demo/full_screen_dialog_demo.dart (hunk 2 above)
Comment (HansMuller) [inline examples/material_gallery/lib/demo/full_screen_dialog_demo.dart]: Oops, that was an old note to myself that I should have deleted.

**Pull request merged**